src/interfaces/IGuardianVerifier.sol

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.25;
+
+/**
+ * @title IGuardianVerifier
+ * @notice Interface for the verifier contract
+ * @dev A standard interface for verification based on several proof scheme
+ *
+ * This contract contains the logic for the verifier initialisation & verification
+ * It's developed to support proof schemes like zkEmail groth16, zkEmail.nr, zk JWT, but not limited to these
+ */
+interface IGuardianVerifier {
+    /*´:°•.°+.*•´.*:˚.°*.˚•´.°:°•.°•.*•´.*:˚.°*.˚•´.°:°•.°+.*•´.*:*/
+    /*                    DATA STRUCTURES                         */
+    /*.•°:°.´+˚.*°.˚:*.´•*.+°.•°:´*.´•*.•°.•°:°.´:•˚°.*°.˚:*.´+°.•*/
+
+    /// @dev Proof data required proof verification
+    struct ProofData {
+        /// @dev Proof
+        bytes proof;
+        /// @dev Public inputs required for verification
+        bytes32[] publicInputs;
+        /// @dev Additional Data required for verification
+        bytes data;
+    }
+
+    /*´:°•.°+.*•´.*:˚.°*.˚•´.°:°•.°•.*•´.*:˚.°*.˚•´.°:°•.°+.*•´.*:*/
+    /*                    FUNCTIONS                               */
+    /*.•°:°.´+˚.*°.˚:*.´•*.+°.•°:´*.´•*.•°.•°:°.´:•˚°.*°.˚:*.´+°.•*/
+
+    /**
+     * @dev Initialize the contract with ERC1967 proxy deployement
+     * Deployer is AccountRecovery module contract
+     *
+     * @param recoveredAccount Account to be recovered
+     * @param accountSalt Account salt
+     * @param initData Initialization data
+     */
+    function initialize(
+        address recoveredAccount,
+        bytes32 accountSalt,
+        bytes calldata initData
```

Comment:
What use cases did you have in mind for this `initData` again?

Reply:
On top of my mind, it could be to setup access restriction based parameters, mostly taking reference from `EmailAuth.sol`

I am not sure about it's user in other cases